File: Selenium/Activities/Activity22.java
package activities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity22 {

	public static void main(String[] args) {

		WebDriver driver =  new FirefoxDriver();
		
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		
		Actions builder = new Actions(driver);
		
		driver.get("https://v1.training-support.net/selenium/popups");
		
		System.out.println(driver.getTitle());
		
		WebElement button = driver.findElement(By.className("orange"));
        
        builder.moveToElement(button).build().perform();
        
        String tooltipMessage = button.getAttribute("data-tooltip");
        System.out.println(tooltipMessage);

        button.click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("username")));

        WebElement username = driver.findElement(By.id("username"));
        WebElement password = driver.findElement(By.id("password"));
        
        username.sendKeys("admin");
        password.sendKeys("password");
        
        driver.findElement(By.xpath("//button[text()='Log in']")).click();

        String message = driver.findElement(By.id("action-confirmation")).getText();
        System.out.println("Login message: " + message);
        
        driver.close();

	}

}

File: Selenium/Activities/Activity22.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait


with webdriver.Firefox() as driver:
    
    driver.get("https://v1.training-support.net/selenium/popups")

    print(driver.title)

    wait = WebDriverWait(driver, 10)

    signin_button = driver.find_element(By.CLASS_NAME, "orange")
   
    webdriver.ActionChains(driver).move_to_element(signin_button).perform()
    
    tooltip_text = signin_button.get_attribute("data-tooltip")
    print("Button tooltip text: ", tooltip_text)
    
    signin_button.click()

    username = driver.find_element(By.ID, "username")
    password = driver.find_element(By.ID, "password")
    
    username.send_keys("admin")
    password.send_keys("password")
    
    driver.find_element(By.XPATH, "/html/body/div[4]/div/div[2]/div/div/button").click()

    message = driver.find_element(By.ID, "action-confirmation")
    print("Confirmation message: ", message.text)

    driver.close()
